### PR TITLE
Added delete slices function

### PR DESCRIPTION
**Pull request by mathturtle**
@yijiang1 Here is my update of #201 that uses a custom widget class to get the range of slices to delete.  I updated the widget from the last version so that the 'Axis' label and dropdown can be optionally hidden.

### CODE REVIEW

**Comment by yijiang1** [comment]
Looks good to me.

File: tomviz/AddPythonTransformReaction.cxx [hunk 1]
```diff
@@ -228,6 +339,46 @@ OperatorPython* AddPythonTransformReaction::addExpression(DataSource* source)
           source->addOperator(op);
       }
   }
+    
+  else if (scriptLabel == "Delete Slices")
+  {
+      vtkTrivialProducer *t = vtkTrivialProducer::SafeDownCast(
+                                                               source->producer()->GetClientSideObject());
```
Comment (cryos) [inline tomviz/AddPythonTransformReaction.cxx]: Not sure of the point of the newline here - doesn't reduce line length below 80 or make it more readable. Typo?

File: tomviz/AddPythonTransformReaction.cxx [hunk 2]
```diff
@@ -228,6 +339,46 @@ OperatorPython* AddPythonTransformReaction::addExpression(DataSource* source)
           source->addOperator(op);
       }
   }
+    
+  else if (scriptLabel == "Delete Slices")
+  {
+      vtkTrivialProducer *t = vtkTrivialProducer::SafeDownCast(
```
Comment (cryos) [inline tomviz/AddPythonTransformReaction.cxx]: This looks over indented

File: tomviz/AddPythonTransformReaction.cxx [hunk 3]
```diff
@@ -228,6 +339,46 @@ OperatorPython* AddPythonTransformReaction::addExpression(DataSource* source)
...
+      connect(buttons, SIGNAL(rejected()), &dialog, SLOT(reject()));
+      v->addWidget(sliceRange);
+      v->addWidget(buttons);
+      dialog.setLayout(v);
+      dialog.setWindowTitle("Delete Slices");
+      dialog.layout()->setSizeConstraint(QLayout::SetFixedSize); //Make the UI non-resizeable
+      
+      if (dialog.exec() == QDialog::Accepted)
+      {
+          QString deleteSlicesScript = scriptSource;
```
Comment (cryos) [inline tomviz/AddPythonTransformReaction.cxx]: Same here, are these all using four spaces to indent?

**Comment by cryos** [comment]
Some whitespace issues, but looks good otherwise. It looks like four spaces in some code for indentation.

**Comment by mathturtle** [comment]
I took @yijiang1's branch from #201 and amended it.  I'd guess his editor is defaulting to 4 spaces for indentation since I didn't modify that part of the code very much.  I've fixed the indentation now.

There are some other indentation issues in that file.  I'll fix them in my 'fix the braces' branch.

**Comment by cryos** [comment]
Sounds good

**Pull request merged**